test/core/security/credentials_test.cc

```diff
@@ -3799,6 +3813,895 @@ TEST(CredentialsTest,
+      "\"audience\":\"audience\","
+      "\"subject_token_type\":\"subject_token_type\","
+      "\"service_account_impersonation_url\":\"service_account_impersonation_"
+      "url\",\"token_url\":\"token_url\","
+      "\"token_info_url\":\"token_info_url\",	\"credential_source\": "
+      "{\"executable\":{"
+      "\"command\":\"command\","
+      "\"timeout_millis\":4999,"
+      "\"output_file\":\"output_file\"}},"
+      "\"quota_project_id\":\"quota_project_id\","
+      "\"client_id\":\"client_id\","
+      "\"client_secret\":\"client_secret\"}";
+  auto options_json = JsonParse(options_string);
+  GPR_ASSERT(options_json.ok());
+  grpc_error_handle error;
+  auto creds =
+      ExternalAccountCredentials::Create(*options_json, {"scope"}, &error);
+  std::string actual_error,
+      expected_error =
+          "timeout_millis should be between 5000 and 120000 milliseconds.";
+  grpc_error_get_str(error, StatusStrProperty::kDescription, &actual_error);
+  GPR_ASSERT(strcmp(actual_error.c_str(), expected_error.c_str()) == 0);
+}
+
+TEST(
+    CredentialsTest,
+    TestPluggableAuthExternalAccountCredentialsCreateFailureExecutableTimeoutMoreThan120Seconds) {
+  const char* options_string =
+      "{\"type\":\"external_account\","
+      "\"audience\":\"audience\","
+      "\"subject_token_type\":\"subject_token_type\","
+      "\"service_account_impersonation_url\":\"service_account_impersonation_"
+      "url\",\"token_url\":\"token_url\","
+      "\"token_info_url\":\"token_info_url\",	\"credential_source\": "
+      "{\"executable\":{"
+      "\"command\":\"command\","
+      "\"timeout_millis\":120001,"
+      "\"output_file\":\"output_file\"}},"
+      "\"quota_project_id\":\"quota_project_id\","
+      "\"client_id\":\"client_id\","
+      "\"client_secret\":\"client_secret\"}";
+  auto options_json = JsonParse(options_string);
+  GPR_ASSERT(options_json.ok());
+  grpc_error_handle error;
+  auto creds =
+      ExternalAccountCredentials::Create(*options_json, {"scope"}, &error);
+  std::string actual_error,
+      expected_error =
+          "timeout_millis should be between 5000 and 120000 milliseconds.";
+  grpc_error_get_str(error, StatusStrProperty::kDescription, &actual_error);
+  GPR_ASSERT(strcmp(actual_error.c_str(), expected_error.c_str()) == 0);
+}
+
+#ifndef GPR_WINDOWS
+
+char* get_pluggable_auth_executable_file_contents_for_output_file(
+    int32_t expiration_time, const char* output_file_filename) {
+  std::string executable_file_contents = absl::StrFormat(
+      "#!/bin/bash\n\necho \"{\\\"version\\\": 1,\\\"success\\\": "
+      "\"true\",\\\"token_type\\\": "
+      "\\\"urn:ietf:params:oauth:token-type:saml2\\\",\\\"saml_response\\\":"
+      "\\\"test_subject_token\\\",\\\"expiration_time\\\":%d}\" > %s",
+      expiration_time, output_file_filename);
+  return gpr_strdup(executable_file_contents.c_str());
+}
+
+char* get_pluggable_auth_executable_file_contents(int32_t expiration_time) {
```

Comment:
This should be something like get_pluggable_auth_executable_saml, and the other get_pluggable_auth_executable_oidc. Why not just create & return the file?